text_mining.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+"""
+YOUR HEADER COMMENT HERE
+
+@author: ISA BLANCETT
+
+"""
+
+# Assumptions:
+# Wikipedia articles of states are up to date
+# Distance between states = google maps distance between states
+# If states are next to each other, their distance is 0
+# Hawaii is considered 5000 miles from Alabama for the sake of this script
+
+import wikipedia
+from collections import Counter
+from itertools import chain
+import matplotlib.pyplot as plt
+
+states = ['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California',
+          'Colorado', 'Connecticut', 'Delaware', 'Florida', 'Georgia (U.S. state)',
+          'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa',
+          'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland',
+          'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri',
+          'Montana', 'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey',
+          'New Mexico', 'New York (state)', 'North Carolina', 'North Dakota', 'Ohio',
+          'Oklahoma', 'Oregon', 'Pennsylvania', 'Rhode Island', 'South Carolina',
+          'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont',
+          'Virginia', 'Washington (state)', 'West Virginia', 'Wisconsin', 'Wyoming']
+
+# In miles
+dist_al = [0, 4303, 1564, 446, 2161, 1444, 1157, 932, 0, 0,
+           5000, 2131, 700, 614, 948, 1009, 500, 388, 1507, 869,
+           1262, 935, 1324, 0, 609, 1964, 1166, 2084, 1325, 1028,
+           1228, 1211, 585, 1620, 728, 728, 2422, 981, 1225, 395,
+           1315, 0, 828, 1773, 1360, 669, 2547, 705, 934, 1650]
+
+freq_words = []
+
+def get_content(title):
+    """ grabs the content of a wikipedia page
+
+        title: name of the wikipedia page
+        returns: content of wiki page in plain text
+
+    >>> get_content('Olin College')
+    'Olin College of Engineering (also known as Olin College or simply Olin) is a private....'
+    """
+    article = wikipedia.page(title)
+    return article.content
+
+def find_freq_words(title):
+    """ inserts word_freq list ordered by frequency & deletes punctuation
+
+        title: name of the wikipedia page
+        calls: get_content(title)
+        returns: nothing
+    """
+
+    words = get_content(title).split()
+
+    # Checks for words carrying punctuation marks
+    for i in range(len(words)):
+        if words[i].endswith('.') or words[i].endswith(',') or words[i].endswith(')') or words[i].endswith(':'):
+            words[i] = words[i][:-1]
+        if words[i].startswith("("):
+            words[i] = words[i][1:]
+        if words[i].endswith("'s"):
+            words[i] = words[i][:-2]
+
+    ordered = sorted(words, key = words.count, reverse=True)
+
+    # Orders by frequency without repeat terms
+    seen = {}
+    ordered_set = [seen.setdefault(x, x) for x in ordered if x not in seen]
+    freq_words.append(ordered_set)
+    return
```

Comment:
This is a void function that uses a global variable. Generally, I would recommend implementing a fruitful function instead that does not rely on a global variable. Please refer to this StackOverflow article to find why https://stackoverflow.com/questions/19158339/why-are-global-variables-evil